lms/views/assignment.py

```diff
@@ -0,0 +1,54 @@
+from pyramid.view import view_config
+from lms.security import Permissions
+from lms.resources._js_config.file_picker_config import FilePickerConfig
+from lms.validation._base import PyramidRequestSchema
+from marshmallow import Schema
+
+from webargs import fields
+
+
+class APISyncSchema(PyramidRequestSchema):
+    class LMS(Schema):
+        product = fields.Str(required=True)
+
+    lms = fields.Nested(LMS, required=True)
+
+
+@view_config(
+    request_method="POST",
+    permission=Permissions.API,
+    renderer="json",
+    route_name="api.assignment.config",
+    schema=APISyncSchema,
+)
+def config(_context, request):
```

Comment:
Not sure about the naming of this. Is it the config of the assignment? 

It is but also it contains the information to reconfigure it :shrug:

Reply:
You could split this into two views, one that returns the existing configuration for the assignment and another that returns the assignment-agnostic data needed by the file-picker app. The frontend could make both requests in parallel.